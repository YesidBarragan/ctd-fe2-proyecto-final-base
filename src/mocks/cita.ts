/* istanbul ignore file */
import { rest } from 'msw';
import { API_URL } from '../app/constants';

export const handlers = [
  rest.get(API_URL, (req, res, ctx) => {

    const homerQuote = [
      {
        quote: "All I'm gonna use this bed for is sleeping, eating and maybe building a little fort.",
        character: "Homer Simpson",
        image:
          "https://cdn.glitch.com/3c3ffadc-3406-4440-bb95-d40ec8fcde72%2FHomerSimpson.png?1497567511939",
        characterDirection: "Right",
      }
    ];

    const randomQuote = [
      {
        quote: "Nothing you say can upset us. We're the MTV generation.",
        character: "Bart Simpson",
        image:
          "https://cdn.glitch.com/3c3ffadc-3406-4440-bb95-d40ec8fcde72%2FBartSimpson.png?1497567511638",
        characterDirection: "Right",
      }
    ];

    if (req.url.searchParams.get('character')) {
      return res(
        ctx.status(200),
        ctx.json(homerQuote)
      );
    }

    return res(
      ctx.status(200),
      ctx.json(randomQuote)
    );

  })
];