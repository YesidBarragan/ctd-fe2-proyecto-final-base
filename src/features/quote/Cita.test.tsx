import userEvent from "@testing-library/user-event";
import { setupServer } from "msw/node";
import { handlers }  from "../../mocks/cita";
import { render, screen, fireEvent } from "../../test-utils";
import Cita from "./Cita";

const server = setupServer(...handlers);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe('Cita component', () => {

  it('should render correctly', () => {
    const { asFragment } = render(<Cita/>);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByPlaceholderText('Ingresa el nombre del autor'));
  })

  it('should change the value of the input', () => {
    render(<Cita/>)
    const searchInput = screen.getByPlaceholderText('Ingresa el nombre del autor');
    fireEvent.change(searchInput, { target: { value: 'Homer' } });
    expect(searchInput as HTMLInputElement).toHaveValue('Homer');
  })

  it('should render a random quote', async () => {
		render(<Cita />);
		const searchButton = screen.getByText('Obtener cita aleatoria');
		await userEvent.click(searchButton);
		expect(await screen.findByText('Bart Simpson')).toBeInTheDocument()
	})

  it('should render a character quote', async () => {
		render(<Cita />);
		const searchInput = screen.getByPlaceholderText("Ingresa el nombre del autor")
		await userEvent.type(searchInput, 'Homer')
		const searchButton = screen.getByText('Obtener Cita');
		await userEvent.click(searchButton);
		expect(await screen.findByText('Homer Simpson')).toBeInTheDocument()
	})

  it('should render a error message', async () => {
		render(<Cita />);
		const searchInput = screen.getByPlaceholderText("Ingresa el nombre del autor")
		await userEvent.type(searchInput, '1')
		const searchButton = screen.getByText('Obtener Cita');
		await userEvent.click(searchButton);
		expect(await screen.findByText('Por favor ingrese un nombre vÃ¡lido')).toBeInTheDocument()
	})

  it('should clear the input', () => {
    render(<Cita/>)
    const searchInput = screen.getByPlaceholderText('Ingresa el nombre del autor');
    fireEvent.change(searchInput, { target: { value: 'Homer' } });
    const clearButton = screen.getByText('Borrar');
    fireEvent.click(clearButton);
    expect(searchInput as HTMLInputElement).toHaveValue('');
  })
})