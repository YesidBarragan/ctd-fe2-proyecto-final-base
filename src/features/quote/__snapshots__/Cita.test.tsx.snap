// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cita component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ ccfNVZ"
  >
    <p
      class="sc-gsnTZi lonubk"
    >
      No se encontro ninguna cita
    </p>
    <p
      class="sc-dkzDqf cfsdtw"
    />
    <input
      aria-label="Author Cita"
      class="sc-hKMtZM ewBjec"
      placeholder="Ingresa el nombre del autor"
      value=""
    />
    <button
      aria-label="Obtener cita aleatoria"
      class="sc-eCYdqJ jqdBt"
    >
      Obtener cita aleatoria
    </button>
    <button
      aria-label="Borrar"
      class="sc-eCYdqJ LyeJx"
    >
      Borrar
    </button>
  </div>
</DocumentFragment>
`;
